clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -1,13 +1,92 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+import com.google.gson.JsonArray;
 import org.junit.Test;
 
 import static org.junit.Assert.*;
 
 public class HubConnectionTest {
-    // Coming soon!
     @Test
-    public void testEmptyCollection() {
+    public void checkHubConnectionState() throws InterruptedException {
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("http://example.com", mockTransport);
+            hubConnection.start();
+            assertTrue(hubConnection.connected);
```

Comment:
I hope `connected` is going to go away in later PRs

Reply:
Planning on replacing this with an enum like the other clients